includes/utilities.inc

```diff
@@ -375,11 +375,13 @@ function islandora_solr_prepare_solr_results($solr_results) {
             // eg: content models.
             else {
               if ($object = islandora_object_load($pid)) {
-                $label = $object->label;
+                if (islandora_object_access(ISLANDORA_VIEW_OBJECTS, $object)) {
+                  $label = $object->label;
+                }
               }
             }
           }
-          return ($label ? $label : $uri);
+          return (isset($label) ? $label : $uri);
```

Comment:
isset is not the same as !not empty. If empty, you will be returning an empty label